Fix first carbon instance for every service

jordiclariana opened the pull request:
The list of INSTANCES variable was something like: "aggregator b c d" or "cache b c d"
First instance should be 'a': "a b c d"
Before fix, first instance was never started, stopped or restarted.

dwerder commented:
It was never started? Sure? All the people out there never realized that their Graphite isn't working?
I don't thing so. Take a look at that line https://github.com/echocat/puppet-graphite/blob/master/templates/etc/init.d/RedHat/carbon-cache.erb#L37 for example.

Your solution is also valid, but it fixes a non-existing problem. We can discuss if its the better solution.

dwerder commented:
But I would merge that PR, if the commit is ready and the old solution is removed. Because your solution is more elegant.

jordiclariana commented:
You are right, it worked on Redhad, but not on Debian (Debian didn't have that line you gave me as example). I was on Debian, realized the problem and fixed it. I just changed the redhat init scripts too because I thought they where mostly the same. My fault.

About the fix itself, you're right, it worked, but it started and stopped a instance like this:
python -W ignore /opt/graphite/bin/carbon-aggregator.py --instance=aggregator start

I'm not sure what the 'aggregator' instance points to, but I guess is the default one, the 'a'. So at least for clarity is better to use 'a'. My problem was that I had a 'a' instance running, upgraded the init scripts, and then the restart didn't stop my 'a' instance.

Regards.
